src/Component/JsonSchema/Guesser/Validator/Format/UriValidator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Jane\Component\JsonSchema\Guesser\Validator\Format;
+
+use Jane\Component\JsonSchema\Guesser\Guess\ClassGuess;
+use Jane\Component\JsonSchema\Guesser\Guess\Property;
+use Jane\Component\JsonSchema\Guesser\Validator\ValidatorGuess;
+use Jane\Component\JsonSchema\Guesser\Validator\ValidatorInterface;
+use Jane\Component\JsonSchema\JsonSchema\Model\JsonSchema;
+use Symfony\Component\Validator\Constraints\Url;
+
+class UriValidator implements ValidatorInterface
+{
+    public function supports($object): bool
+    {
+        return $object instanceof JsonSchema && ((\is_array($object->getType()) ? \in_array('string', $object->getType()) : 'string' === $object->getType()) || null === $object->getType()) && \in_array($object->getFormat(), ['uri', 'uri-reference']);
+    }
+
+    /**
+     * @param JsonSchema          $object
+     * @param ClassGuess|Property $guess
+     */
+    public function guess($object, string $name, $guess): void
+    {
+        $guess->addValidatorGuess(new ValidatorGuess(Url::class, [
```

Comment:
An URL is an URI. But not all URIs are URLs.
Looking at `Symfony\Component\Validator\Constraints::$protocols`, it's only URL and limited to `http` and `https` =/

Reply:
I tried with a regex I found but the regex wasn't supported by PHP regex engine so this is the best I could find :/ If you have a regex that works I'm up to change that !

Reply:
Didn't find a specific Regex; try look at https://github.com/thephpleague/uri/blob/master/src/UriString.php
Might be redundant with previous link for IRI.